### ISSUE TITLE
hang in signal/test_async_timer_mt.multithread/0

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#413]
(Mellanox internal link) http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/349/label=hpc-test-node/consoleFull

**Comment by yosefe** [comment]
fixes in #695

**Comment by yosefe** [comment]
happens again in http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3283/label=hpc-arm-04,worker=2/console
```
14:21:53 [----------] 1 test from signal/test_async_timer_mt
14:21:53 [ RUN      ] signal/test_async_timer_mt.multithread/0
16:00:29 Build timed out (after 120 minutes). Marking the build as failed.
16:00:29 Build was aborted
16:00:29 TAP Reports Processing: START
16:00:29 Looking for TAP results report in workspace using pattern: **/*.tap
16:00:29 Saving reports...
16:00:29 Processing '/scrap/jenkins/jobs/hpc-ucx-pr/configurations/axis-label/hpc-arm-04/axis-worker/2/builds/3283/tap-master-files/build-test/mpi_tests.tap'
16:00:29 Parsing TAP test result [/scrap/jenkins/jobs/hpc-ucx-pr/configurations/axis-label/hpc-arm-04/axis-worker/2/builds/3283/tap-master-files/build-test/mpi_tests.tap].
16:00:29 TAP Reports Processing: FINISH
16:00:29 Archiving artifacts
16:00:29 
make: *** [test] Terminated
```
